lib/p2p/Pool.ts

```diff
@@ -65,16 +69,21 @@ class Pool extends EventEmitter {
     this.connected = false;
   }
 
-  public addOutbound = async (address: string, port: number): Promise<Peer> => {
-    const socketAddress = new SocketAddress(address, port);
+  /**
+   * Attempt to add an outbound peer by connecting to a given SocketAddress.
+   * Throws an error if a connection already exists for the provided address.
+   * If the connection is successful, send a [[GetHostsPacket]] to discover new XUD hosts to connect to.
+   */
+  public addOutbound = async (socketAddress: SocketAddress): Promise<Peer> => {
     if (this.peers.has(socketAddress)) {
-      const err = errors.ADDRESS_ALREADY_CONNECTED(address.toString());
+      const err = errors.ADDRESS_ALREADY_CONNECTED(socketAddress.address);
       this.logger.info(err.message);
       throw err;
     }
 
     const peer = Peer.fromOutbound(socketAddress);
     await this.tryConnectPeer(peer);
+    peer.sendPacket(new GetHostsPacket(null));
```

Comment:
This should be placed in `handleOpen`, with all the other post-handshake operations. 
At the moment it also doesn't apply for inbound connections. Was that on purpose?

Reply:
Yes that was intentional, although I could make it symmetrical. In that case though, should we just have this be part of the handshake process?

Reply:
Handshake process is for critical stuff, to block other activities (same as if the peer is not connected yet). So this should be post-handshake.

Reply:
OK, that makes sense to me just wanted to make sure. Another idea, maybe for a future PR, is that we might want a configuration option to determine whether or not we automatically connect to new hosts that we learn about through getHosts. One might want manual control over which peers they connect to, although it's harmless to at least learn about new peers.

Reply:
Possible, but remember that the new design (#185) would require to save hosts to list before attempting to connect to them.

@kilrau